pom.xml

```diff
@@ -3,284 +3,386 @@
+                                    </banDuplicateClasses>
+                                </rules>
+                            </configuration>
+                        </execution>
+                        <execution>
+                            <id>enforce-ban-version-downgrades</id>
+                            <goals>
+                                <goal>enforce</goal>
+                            </goals>
+                            <configuration>
+                                <rules>
+                                    <requireUpperBoundDeps>
+                                        <uniqueVersions>true</uniqueVersions>
+                                    </requireUpperBoundDeps>
+                                </rules>
+                            </configuration>
+                        </execution>
+                    </executions>
+                    <dependencies>
+                        <dependency>
+                            <groupId>org.codehaus.mojo</groupId>
+                            <artifactId>extra-enforcer-rules</artifactId>
+                            <version>1.0-alpha-3</version>
+                        </dependency>
+                    </dependencies>
+                </plugin>
+
+                <!-- Scala Maven Plugin -->
+                <plugin>
+                    <groupId>net.alchim31.maven</groupId>
+                    <artifactId>scala-maven-plugin</artifactId>
+                    <version>3.2.2</version>
+                    <executions>
+                        <execution>
+                            <id>scala-compile</id>
+                            <phase>process-resources</phase>
+                            <goals>
+                                <goal>add-source</goal>
+                                <goal>compile</goal>
+                            </goals>
+                        </execution>
+                        <execution>
+                            <id>scala-test-compile</id>
+                            <phase>process-test-resources</phase>
+                            <goals>
+                                <goal>testCompile</goal>
+                            </goals>
+                        </execution>
+                    </executions>
+                    <configuration>
+                        <recompileMode>all</recompileMode>
+                        <useZincServer>true</useZincServer>
+                        <scalaVersion>${scala.version}</scalaVersion>
+                        <displayCmd>true</displayCmd>
+                        <verbose>true</verbose>
+                        <failOnMultipleScalaVersions>true</failOnMultipleScalaVersions>
+                        <args>
+                            <arg>-target:jvm-${scala.java.version}</arg>
+                            <arg>-feature</arg>
+                            <arg>-deprecation</arg>
+                            <!-- arg>-Ycheck:jvm</arg -->
```

Comment:
It was to suppress some benign warnings that occurred, and that are fixed in scala 2.11.

It is better to leave this on though, I had just grown weary of seeing them on the screen :)